stl/inc/atomic

```diff
@@ -418,21 +418,42 @@ void _Atomic_wait_direct(
 }
 #endif // _HAS_CXX20
 
-#if 1 // TRANSITION, ABI
+#if 1 // TRANSITION, ABI, GH-1151
 inline void _Atomic_lock_acquire(long& _Spinlock) noexcept {
-    while (_InterlockedExchange(&_Spinlock, 1)) {
-        _YIELD_PROCESSOR();
+#if defined(_M_IX86) || defined(_M_X64)
+    // Algorithm from Intel(R) 64 and IA-32 Architectures Optimization Reference Manual, September 2019
+    // Example 2-4. Contended Locks with Increasing Back-off Example - Improved Version, page 2-21

```

Comment:
I think we need to obtain the proper blessings for this.

Reply:
Moved `atomic,cpp` part to #1200

Reply:
I'm not sure we need anything specific; the "algorithm" / what this is doing is indeed from the optimization manual in the sense that it asks to use increasing numbers of `_mm_pause` over time, but to me that's on the same level as saying 'std::vector is a dynamic array that reallocates when it runs out of space'.

Reply:
Word from Intel is that all code in the pertinent manual is covered by [0BSD](https://opensource.org/licenses/0BSD). We've added a comment annotating that fact.